### PR TITLE
Update features & improve feature output

### PR DESCRIPTION
**Pull request by torotil**

### CODE REVIEW

File: modules/campaignion_donation_type/campaignion_donation_type.features.inc [hunk 1]
```diff
@@ -48,7 +48,7 @@ function campaignion_donation_type_node_info() {
 function campaignion_donation_type_default_payment_method() {
   $items = array();
   $items['manual_direct_debit'] = entity_import('payment_method', '{
-    "controller_data" : [],
+    "controller_data" : { "day_options" : [ "1", "15", "28" ], "long_account_numbers" : true },
```
Comment (katzmo) [inline modules/campaignion_donation_type/campaignion_donation_type.features.inc:51]: Are those the default settings for the payment method (would the empty array before have resulted in the same settings)?

File: modules/campaignion_conf/campaignion_conf.module [hunk 2]
```diff
@@ -12,6 +12,14 @@ function campaignion_conf_module_implements_alter(&$implementations, $hook) {
     unset($implementations['image']);
     unset($implementations['media_youtube']);
   }
+  if ($hook == 'features_pipe_node_alter') {
+    // Ignore diff & comment variables for node-types.
```
Comment (katzmo) [inline modules/campaignion_conf/campaignion_conf.module]: Is this comment in the right place? I find it rather confusing TBH.

A comment about filtering comment variables in `campaignion_conf_features_pipe_node_alter` on the other hand would be nice.

File: modules/campaignion_donation_type/campaignion_donation_type.info [hunk 3]
```diff
@@ -26,8 +26,10 @@ dependencies[] = node
 dependencies[] = node_reference
 dependencies[] = opengraph_meta_upload
 dependencies[] = payment
+dependencies[] = pgbar
 dependencies[] = pgbar:pgbar
 dependencies[] = pgbar:pgbar_webform
```
Comment (katzmo) [inline modules/campaignion_donation_type/campaignion_donation_type.info:30]: Does it really need a dependency on `pgbar`, `pgbar:pgbar` and `pbar:pgbar_webform`? Surly the latter depend on pgbar already?!

File: modules/campaignion_context/campaignion_context.info [hunk 4]
```diff
@@ -7,6 +7,8 @@ dependencies[] = cck_blocks
 dependencies[] = context
 dependencies[] = ctools
 dependencies[] = recent_supporters
+dependencies[] = share_light
+dependencies[] = views
```
Comment (katzmo) [inline modules/campaignion_context/campaignion_context.info:12]: What about (some form of) pgbar?

File: modules/campaignion_flexible_form/campaignion_flexible_form_templates/campaignion_flexible_form_templates.features.uuid_node.inc [hunk 5]
```diff
@@ -269,8 +269,12 @@ function campaignion_flexible_form_templates_uuid_features_default_content() {
     ),
     'conditionals' => array(),
   ),
-  'date' => '2012-07-23 16:10:30 +0200',
-  'user_uuid' => '59e16a84-5bfa-438b-be8f-782a0af74750',
+  'path' => array(
+    'pathauto' => FALSE,
+    'alias' => 'content/default-flexible-form-template-do-not-delete',
+  ),
```
Comment (katzmo) [inline modules/campaignion_flexible_form/campaignion_flexible_form_templates/campaignion_flexible_form_templates.features.uuid_node.inc:275]: The pathauto settings seem a bit inconsistent to me, shouldn't it be either TRUE or FALSE for *all* content types? 
Also is the module always enabled?

File: modules/campaignion_petition/campaignion_petition_templates/campaignion_petition_templates.features.uuid_node.inc [hunk 6]
```diff
@@ -10,6 +10,61 @@
...
+    ),
+  ),
+  'path' => array(
+    'pathauto' => FALSE,
+    'alias' => 'content/thank-you-you’re-awesome',
+  ),
+  'date' => '2017-04-07 11:24:30 +0000',
+  'user_uuid' => '9dbdd05e-9e99-4f34-a31b-8685d3f6bcc3',
+  'webform' => NULL,
+);
```
Comment (katzmo) [inline modules/campaignion_petition/campaignion_petition_templates/campaignion_petition_templates.features.uuid_node.inc:67]: I thought you wanted to remove the exported thank you pages? ;)

File: modules/campaignion_donation_type/campaignion_donation_type.features.inc (hunk 1 above)
Comment (torotil) [reply modules/campaignion_donation_type/campaignion_donation_type.features.inc:51]: I think so, yes.

File: modules/campaignion_conf/campaignion_conf.module (hunk 2 above)
Comment (torotil) [reply modules/campaignion_conf/campaignion_conf.module]: Yes it’s in the right place. Removing the diff-hook is all that needs to be done for removing the diff variables.

Comment (torotil) [reply modules/campaignion_conf/campaignion_conf.module]: Comments is more complicated as it is added by strongarm together with other variables (that we do need). So the purpose of the whole section in between the curly-praces is exactly what’s stated. That‘s the “why?” for this block of code.

File: modules/campaignion_donation_type/campaignion_donation_type.info (hunk 3 above)
Comment (torotil) [reply modules/campaignion_donation_type/campaignion_donation_type.info:30]: Ah that was accidental, need to remove that. Thanks.

File: modules/campaignion_context/campaignion_context.info (hunk 4 above)
Comment (torotil) [reply modules/campaignion_context/campaignion_context.info:12]: I haven’t added these manually. I wonder why pgbar wasn’t pulled in if we have a related context reaction.

File: modules/campaignion_conf/campaignion_conf.module (hunk 2 above)
Comment (katzmo) [reply modules/campaignion_conf/campaignion_conf.module]: But the comment is *inside* the curly brackets so it just comments `unset($implementations['diff']);` which has nothing to do with comment variables at all, so IMHO the comment should only say "Ignore diff variables for node-types.". The filtering comment variables part is actually in the hook that's just being moved to being called as late as possible here.

File: modules/campaignion_petition/campaignion_petition_templates/campaignion_petition_templates.features.uuid_node.inc (hunk 6 above)
Comment (torotil) [reply modules/campaignion_petition/campaignion_petition_templates/campaignion_petition_templates.features.uuid_node.inc:67]: I didn’t suggest to do it in this PR though.

File: modules/campaignion_donation_type/campaignion_donation_type.info (hunk 3 above)
Comment (torotil) [reply modules/campaignion_donation_type/campaignion_donation_type.info:30]: Updated the commit to exclude the additional dependency.

File: modules/campaignion_conf/campaignion_conf.module (hunk 2 above)
Comment (torotil) [reply modules/campaignion_conf/campaignion_conf.module]: I’ve moved the comment one line up.

File: modules/campaignion_flexible_form/campaignion_flexible_form_templates/campaignion_flexible_form_templates.features.uuid_node.inc (hunk 5 above)
Comment (torotil) [reply modules/campaignion_flexible_form/campaignion_flexible_form_templates/campaignion_flexible_form_templates.features.uuid_node.inc:275]: I’ve made it FALSE in all nodes.

File: modules/campaignion_context/campaignion_context.info (hunk 4 above)
Comment (torotil) [reply modules/campaignion_context/campaignion_context.info:12]: I’ve added it manually. Still strange that features doesn’t find this automatically :shrug: I think it’s missing a proper support for cck_blocks.

File: modules/campaignion_conf/campaignion_conf.module (hunk 2 above)
Comment (katzmo) [reply modules/campaignion_conf/campaignion_conf.module]: What about this? More consistency, less duplicate code:

```php
/**
 * Implements hook_module_implements_alter().
 */
function campaignion_conf_module_implements_alter(&$implementations, $hook) {
  if ($hook == 'file_default_displays_alter') {
    unset($implementations['image']);
    unset($implementations['media_youtube']);
  }
  // Ignore diff variables for node-types.
  if ($hook == 'features_pipe_node_alter') {
    unset($implementations['diff']);
  }
  // Ensure our hook implementations are called late.
  if (in_array($hook, ['features_pipe_node_alter', 'features_pipe_context_alter'])) {
    $group = $implementations['campaignion_conf'];
    unset($implementations['campaignion_conf']);
    $implementations['campaignion_conf'] = $group;
  }
}

/**
 * Implements hook_features_pipe_COMPONENT_alter() for context.
 */
function campaignion_conf_features_pipe_context_alter(&$pipe, $data, $export) {
  // Deactivate content-type dependencies caused by context conditions.
  unset($pipe['node']);
}

/**
 * Implements hook_features_pipe_COMPONENT_alter() for node.
 */
function campaignion_conf_features_pipe_node_alter(&$pipe, $data, $export) {
  // Ignore comment variables for node-types.
  $pipe['variable'] = array_filter($pipe['variable'], function ($name) {
    if (substr($name, 0, 8) == 'comment_') {
      return FALSE;
    }
    return TRUE;
  });
}
```

Comment (torotil) [reply modules/campaignion_conf/campaignion_conf.module]: Ok, will change it accordingly …

Comment (torotil) [reply modules/campaignion_conf/campaignion_conf.module]: Or not :thinking: The reason for this that I usually want to know what happens to each hook. I can see that more clearly with the previous version, although it has more code duplication.

Comment (torotil) [reply modules/campaignion_conf/campaignion_conf.module]: How about factoring it out like this?

Comment (katzmo) [reply modules/campaignion_conf/campaignion_conf.module]: Ok, then (edited for update)

```php
/**
 * Implements hook_module_implements_alter().
 */
function campaignion_conf_module_implements_alter(&$implementations, $hook) {
  // Ensure our hook implementations are called late.
  $move_to_the_end = function (&$implementations, $module) {
    $group = $implementations[$module];
    unset($implementations[$module]);
    $implementations[$module] = $group;
  };
  if ($hook == 'file_default_displays_alter') {
    unset($implementations['image']);
    unset($implementations['media_youtube']);
  }
  if ($hook == 'features_pipe_context_alter') {
    $move_to_the_end($implementations, 'campaignion_conf');
  }
  if ($hook == 'features_pipe_node_alter') {
    $move_to_the_end($implementations, 'campaignion_conf');
    // Ignore diff variables for node-types.
    unset($implementations['diff']);
  }
}

/**
 * Implements hook_features_pipe_COMPONENT_alter() for context.
 */
function campaignion_conf_features_pipe_context_alter(&$pipe, $data, $export) {
  // Deactivate content-type dependencies caused by context conditions.
  unset($pipe['node']);
}

/**
 * Implements hook_features_pipe_COMPONENT_alter() for node.
 */
function campaignion_conf_features_pipe_node_alter(&$pipe, $data, $export) {
  // Ignore comment variables for node-types.
  $pipe['variable'] = array_filter($pipe['variable'], function ($name) {
    if (substr($name, 0, 8) == 'comment_') {
      return FALSE;
    }
    return TRUE;
  });
}
```

Comment (katzmo) [reply modules/campaignion_conf/campaignion_conf.module]: :+1: for factoring out, still not happy about comments :wink: 
(updated suggestion above)

Comment (torotil) [reply modules/campaignion_conf/campaignion_conf.module]: Hm, now the comment before `$move_to_the_end` is lying unless I hardcode `$module='campaignion_conf'`. Can I just remove the comment? [I think the variable name speaks for itself.]

Comment (katzmo) [reply modules/campaignion_conf/campaignion_conf.module]: Why, wouldn't it be called late for any module? :thinking: But fine with not commenting the function.

Could you also fix the indentation and maybe add an overall function comment? Thanks!

Comment (torotil) [reply modules/campaignion_conf/campaignion_conf.module]: … and also fixed the wrong intentation. If not I’d suggest use some real-time code collaboration for this …

Comment (katzmo) [reply modules/campaignion_conf/campaignion_conf.module]: Looks perfect now! 

Except for the missing doc comment but that's your domain usually…

**Review by katzmo — APPROVED** [review]

**Pull request merged**